tests/queries/0_stateless/02590_interserver_mode_client_info_initial_query_start_time.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+# Tags: no-fasttest
+# Tag no-fasttest: interserver mode requires SSL
+#
+# Test that checks that some of ClientInfo correctly passed in inter-server mode.
+# NOTE: we need .sh test (.sql is not enough) because queries on remote nodes does not have current_database = currentDatabase()
+#
+# Check-style suppression: select * from system.query_log where current_database = currentDatabase();
+
+CUR_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CUR_DIR"/../shell_config.sh
+
+function get_query_id() { random_str 10; }
+
+$CLICKHOUSE_CLIENT -nm -q "
+    drop table if exists buf;
+    drop table if exists dist;
+    drop table if exists data;
+
+    create table data (key Int) engine=Memory();
+    create table dist as data engine=Distributed(test_interserver_cluster, currentDatabase(), data, key);
+    create table dist_dist as data engine=Distributed(test_interserver_cluster, currentDatabase(), dist, key);
+    system stop distributed sends dist;
+"
+
+echo "SELECT"
+query_id="$(get_query_id)"
+# initialize connection, but actually if there are other tables that uses this
+# cluster then, it will be created long time ago, but this is OK for this
+# test, since we are care about the difference between NOW() and there should
+# not be any significant difference.
+$CLICKHOUSE_CLIENT --prefer_localhost_replica=0 --query_id "$query_id" -q "select * from dist"
+$CLICKHOUSE_CLIENT -nm --param_query_id "$query_id" -q "
+    system flush logs;
+    select count(), countIf(initial_query_start_time_microseconds != query_start_time_microseconds), countIf(event_time - initial_query_start_time > 1) from system.query_log where type = 'QueryFinish' and initial_query_id = {query_id:String};
```

Comment:
Would not it be flaky in case of 1 second lag of CI?

Reply:
This queries should be issued immediately, one, after another, but on CI under heavy load this can likely takes more time, will adjust to 3 seconds.